tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashService.java

```
@@ -1,12 +1,97 @@
+    private static final String WEEKLY_PERIOD = "weekly";
+    private static final String MONTHLY_PERIOD = "monthly";
+    private static final String TRASH_CLEANUP_ENABLED_SETTING = "trash.cleanup.enabled";
+    private static final String TRASH_CLEANUP_PERIOD_SETTING = "trash.cleanup.period";
+    private final UsersRepository usersRepository;
+    private final JmapSettingsRepository jmapSettingsRepository;
+    private final StoreMailboxManager storeMailboxManager;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    public CleanupTrashService(UsersRepository usersRepository, JmapSettingsRepository jmapSettingsRepository, StoreMailboxManager storeMailboxManager, SessionProvider sessionProvider) {
+        this.usersRepository = usersRepository;
+        this.jmapSettingsRepository = jmapSettingsRepository;
+        this.storeMailboxManager = storeMailboxManager;
+        this.sessionProvider = sessionProvider;
+    }
 
     public Mono<Task.Result> cleanupTrash() {
-        return Mono.just(Task.Result.COMPLETED);
+        return Flux.from(usersRepository.listReactive())
+            .flatMap(this::checkSettingAndCleanupTrashForSingleUser)
+            .reduce(Task.Result.COMPLETED, Task::combine);
+    }
+
+    private Mono<Task.Result> checkSettingAndCleanupTrashForSingleUser(Username username) {
+        return Mono.from(jmapSettingsRepository.get(username))
+            .flatMap(jmapSettings -> getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_ENABLED_SETTING)
+                .map(trashCleanupEnabled -> {
+                    if ("true".equals(trashCleanupEnabled.value())) {
+                        return getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING)
+                            .map(trashCleanupPeriod -> cleanupTrashForSingleUser(username, trashCleanupPeriod.value()))
+                            .orElseThrow(() -> new RuntimeException("The setting value of trash.cleanup.period does not exist"));
```

You should rather return an error Mono this is more idiomatic of reactive programming / functional programming.